db/version_edit.cc

```diff
@@ -55,6 +55,8 @@ enum Tag : uint32_t {
   kDbId,
   kBlobFileAddition,
   kBlobFileGarbage,
+  kWalAddition,
+  kWalDeletion,
```

Comment:
Yes, they will be logged during `PurgeObsoleteFiles`.
The consideration here is that, even if a log has been obsolete due to flush, if `PurgeObsoleteFiles` hasn't deleted/archived the log, the log should still be expected to exist on disk. During recovery checking, we check against the on-disk log files.

Reply:
The idea is that the version edits related to WAL track the state of the on-disk logs, instead of the logical active WALs. The purpose of the tracked on-disk WAL state is to check the on-disk logs during recovery.